samples/cellular/modem_shell/src/sock/sock_shell.c

```diff
@@ -111,6 +111,10 @@ static const char sock_send_usage_str[] =
 	"                                010203040506070809101112\n"
 	"                                01 02 03 04 05 06 07 08 09 10 11 12\n"
 	"                                01020304 05060708 09101112\n"
+	"  -W, --wait_ack            Request a blocking send operation until the data\n"
+	"                            has been sent on-air and acknowledged by the peer,\n"
+	"                            if required by the network protocol. This is not valid\n"
+	"                            when -B0 (non-blocking mode) is used.\n"
```

Comment:
It turns out that `-B0 -W` correctly gives an error in send(), but this is camouflaged by not printing error from send in non-blocking mode and **gives an infinite loop** when using the `-l` option. I think other errors from send also could give the same loop without being noticed.

I can either add a check for `-B0 -W` or fix the infinite loop issue (which can happen in other cases).

Reply:
I did the first option for this PR.